### DISCUSSION TITLE
Conan Center has stopped receiving updates for Conan 1.x packages - only Conan 2.0 is supported moving forward

### DISCUSSION THREAD

**Discussion by jcar87** [conan-io/conan-center-index#25461] [Announcements]
> [!WARNING]  
> **Conan 1.x users**: The legacy Conan Center remote at https://center.conan.io is frozen and will no longer receive updates.
>
> **Conan 2.x users**: Moving forward, updates to recipes and packages are published to the new Conan Center remote, please see the [documentation](https://github.com/conan-io/conan-center-index?tab=readme-ov-file#configure-the-conan-center-remote) for details on how to update your configuration. 

Conan 2.0 was released in February 2022 after years in development. Since then, thousands of users and teams have 
adopted Conan 2.0, and the Conan Center Index repository has continued to see significant growth. In 2023, we received 
close to 6000 Pull Requests for recipes. 

Since the Conan 2.0 release, and to facilitate the upgrade path for our users, we adopted the decision to make recipes 
in Conan Center compatible with both Conan 1.x and Conan 2.0.

Porting all recipes to be Conan 2.0 compatible has been a significant effort - aided by community contributors to which 
we are very grateful. This has placed additional pressure on all resources, including our CI compute capabilities. This 
has also given the team valuable insights into how to improve and streamline the CI and review processes.

On Conan 2 release day - we had published binaries for the top 120 most used recipes. Today, nearly 97% of all recipes 
in Conan Center are compatible with Conan 2.

As time has progressed, today the majority of requests to the Conan Center remote originate from Conan 2 users.

**Today we are announcing that from 4 November 2024, new recipe updates in Conan Center will only be compatible 
with Conan 2. Please read below to see how this might affect you.**

### Users accessing the ``https://center.conan.io`` remote via the Conan client

The Conan Center remote will stop being updated for Conan 1.x users. This means that from this date:
- **No new recipe revisions** will be published or visible to the Conan 1.x client.
- **No new versions of libraries** will be published or made available for Conan 1.x.
- **No new recipes** will be published or visible for Conan 1.x.
- All **previously available and published recipes** and binaries will continue to be accessible.
- **No configuration changes** are required. After the stated deadline, the Conan Center remote for Conan 1.x users 
will be frozen at its current state and will not receive any further updates, as detailed above.
- Under **exceptional circumstances**, updates may be published for security reasons.

### Users maintaining their own fork of the conan-center-index git repository

- There will be a branch that contains the **frozen versions/revisions** on the aforementioned date - details will follow shortly
- From then on, we will **no longer guarantee that recipes remain compatible with Conan 1.x** in the main (master) branch.

## For Conan 2.x users

> [!IMPORTANT]  
> Starting 4 November 2024, all updates to recipes and packages are published to the new Conan Center remote
> Please see the [readme](https://github.com/conan-io/conan-center-index?tab=readme-ov-file#configure-the-conan-center-remote) for details.

From the date of the switch, new recipe updates will be uploaded internally to a new repository. New details will be 
announced soon, the team is working in two possible solutions to facilitate this transition:
- Requests coming from Conan 2 clients will be able to see the up-to-date revisions, without impacting legacy Conan 1, 
and without requiring any configuration changes for Conan 1 users.
- A new remote endpoint will be made public

## Contributions in the conan-center-index repository

- Coinciding with the switch, the Conan Center CI service will only validate recipes and publish packages only for Conan 2
- It will no longer be a requirement to retain Conan 1.x compatibility - contributors are free to use Conan 2-only 
features where it aligns with the PR goal
- **We emphatically ask community contributors to not submit pull requests to perform refactorings to remove old logic or 
mass recipe modernizations**. We want to avoid overwhelming our CI resources and the review team - and continue 
prioritizing new features, bug fixes, new versions, new platforms and new recipes.
  - Pull requests with recipe modernizations only and no functional changes or bug fixes will not be prioritized, and 
we may ask the contributor to reopen the PR at a later date.
  - Pull requests that perform recipe modernizations alongside other valid changes (functional or bug fixes) may 
experience delays in reviews if the changesets/diffs make it less obvious what is being changed.
  - As usual, opening a large number of PRs in a short time is hugely discouraged and against our guidelines.
- The Conan Center CI service will undergo upgrades to rely on Conan 2 features - further changes will be announced in the near future

For questions or feedback, please participate in the discussion below.

**Comment by Andreysizko05** [comment]
Tbh, for me conan 1.x is still better, cause it has less issues with cmake, native work with msvc/gcc. Our recipes and cmakelists are longer for conan 2.x, cause they must consist "fix" for new issues which version 2.x update bring with itself. And some important libs still haven`t recipes for conan 2.x, for example GST plugins. Also I might be wrong, but as I see when new versions of QT or other libs get release, they firstly have recipe for conan 1.x and after small delay have conan 2.x support

**Reply by jcar87** [reply]
Hi @Andreysizko05  thank you for your message.

> Our recipes and cmakelists are longer for conan 2.x

If there are any issues reported for this in the [Conan](https://github.com/conan-io/conan) repository, please let us know so that we an have a look and ensure upgrade paths are prioritized. Our experience is that the Conan 2.x integrations tend to require _less_ CMake boilerplate and the goal is to achieve compatibility without having to modify projects that conform to CMake's modern recommendations. 

> And some important libs still haven`t recipes for conan 2.x, for example GST plugins.

We have a large backlog and updates to the remaining recipes are in contention with all other maintenance tasks - however we will ensure we can move forward this this as soon as possible.

> Also I might be wrong, but as I see when new versions of QT or other libs get release, they firstly have recipe for conan 1.x and after small delay have conan 2.x support

CI validation and package promotion has always happened simultaneously - is the small delay related to when packages are visible in the remote?

**Comment by narcdev** [comment]
Hi, we really would like to move our projects to conan 2.x ! 

But, actually we can't, because we're stuck with this issue : https://github.com/conan-io/conan/issues/14055

**Comment by scandyna** [comment]
I also would like to move to Conan 2.
But, I can't because GitLab does not support it now: https://gitlab.com/groups/gitlab-org/-/epics/6816

**Comment by jonas-singe-bruker** [comment]
Painpoints for my project regarding migration to conan 2:
* missing prebuilt debug binaries for windows (this is so convenient with conan 1!)
* missing [imports] in conanfile.txt. In conan 2 I have to switch to conanfile.py and write my own python code (as per migration guide) or familiarize with the concept of deployers, where runtime_deployer ( https://github.com/conan-io/conan/issues/15381 ) is not documented yet. The old solution was so easy and simple!
* conan 2 support in https://github.com/conan-io/cmake-conan was still experimental and incomplete when conan 2 came out (I didn't reevaluate though for some time)
